girder/utility/filesystem_assetstore_adapter.py

```diff
@@ -106,6 +107,7 @@ def __init__(self, assetstore):
         # happens to existing assetstores that no longer can access their temp
         # directories.
         self.tempDir = os.path.join(self.assetstore['root'], 'temp')
+        self.deleteLock = threading.Lock()
```

Comment:
> We can use a file lock, instead. This is heavier, of course.

Probably the best idea, since that uses the resource that is explicitly shared across all servers, and we have atomic primitives to use with the filesystem.

> Do we have deployed instances with multiple app servers? Do we test for other resource contention issues?

External users have deployed on elastic beanstalk with multiple app servers, but we do not test these sorts of things.